src/simulation/simulation_problem_results.jl

```diff
@@ -529,148 +493,200 @@ function _read_optimizer_stats(res::SimulationProblemResults, store::SimulationS
+    )
+
+    return RealizedMeta(
+        result_initial_time,
+        resolution,
+        count,
+        start_offset,
+        end_offset,
+        interval_len,
+    )
+end
+
+function get_realization(
+    result_values::Dict{
+        OptimizationContainerKey,
+        SortedDict{Dates.DateTime, DataFrames.DataFrame},
+    },
+    meta::RealizedMeta,
+    timestamps,
+)
+    realized_values = Dict{OptimizationContainerKey, DataFrames.DataFrame}()
+    for (key, result_value) in result_values
+        results_concat = Dict{Symbol, Vector{Float64}}()
+        for (step, (t, df)) in enumerate(result_value)
+            first_id = step > 1 ? 1 : meta.start_offset
+            last_id =
+                step == meta.count ? meta.interval_len - meta.end_offset : meta.interval_len
+            result_length = length(first_id:last_id)
+            for colname in propertynames(df)
+                colname == :DateTime && continue
+                # TODO DT: may not be correct
+                col = df[!, colname][first_id:last_id]
+                if !haskey(results_concat, colname)
+                    results_concat[colname] = col
+                else
+                    results_concat[colname] = vcat(results_concat[colname], col)
+                end
+            end
+        end
+        realized_values[key] = DataFrames.DataFrame(results_concat, copycols = false)
+        DataFrames.insertcols!(realized_values[key], 1, :DateTime => timestamps)
+    end
+    return realized_values
+end
+
+"""
+Return the final values for the requested variable names for each time step for a problem.
+Accepts a vector of names for the return of the values. If the time stamps and names are
+loaded using the [load_results!](@ref) function it will read from memory.
+
+# Arguments
+- `names::Vector{Symbol}` : names of desired results
+- `initial_time::Dates.DateTime` : initial time of the requested results
+- `len::Int`: length of results
+"""
+function read_realized_variables(res::SimulationProblemResults; kwargs...)
+    return read_realized_variables(res, collect(keys(res.variable_values)); kwargs...)
+end
+
+function read_realized_variables(res::SimulationProblemResults, variables; kwargs...)
+    return read_realized_variables(res, [VariableKey(x...) for x in variables]; kwargs...)
+end
+
+function read_realized_variables(
+    res::SimulationProblemResults,
+    variables::Vector{<:OptimizationContainerKey};
+    initial_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Int, Nothing} = nothing,
+)
+    meta = RealizedMeta(res, initial_time = initial_time, len = len)
+    timestamps = _process_timestamps(res, meta.initial_time, meta.count)
+    result_values = _read_variables(res, variables, timestamps, nothing)
+    timestamps = get_realized_timestamps(
+        res,
+        initial_time = initial_time,
+        len = meta.count * meta.interval_len,
+    )
+    return get_realization(result_values, meta, timestamps)
```

Comment:
Is this actually correct?

Reply:
What does `get_realization` do?  I don't recall

Reply:
`get_realization` is the code that stitches the final result for each timestamp in a range into a contiguous time series